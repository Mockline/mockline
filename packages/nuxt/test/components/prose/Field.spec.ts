import { describe, it, expect } from 'vitest'
// @ts-ignore
import { MField } from '#components'
import type { TypeOf } from "zod";
import ComponentRender from "../../component-render";

describe('CodeIcon', () => {
  const props = {
    name: 'analytics',
    type: 'string',
    required: false,
  }
  it.each([
    ['with name', { props: { name: 'analytics' } }],
    ['with type', { props: { type: 'string' } }],
    ['with required', { props: { required: true } }],
    ['with all props', { props: { ...props } }],
  ])('renders %s correctly', async (nameOrHtml, options: TypeOf<typeof MField.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MField)
    expect(html).toMatchSnapshot()
  })
})
