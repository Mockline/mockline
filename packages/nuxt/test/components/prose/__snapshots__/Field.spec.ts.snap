// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CodeIcon > renders with all props correctly 1`] = `
"<div class="my-5">
  <div class="flex items-center gap-3 font-mono text-sm"><span class="font-semibold text-(--color-accent)">analytics</span>
    <div class="flex-1 flex items-center gap-1.5 text-xs"><span class="rounded-(--base-radius) bg-(--color-bg-tertiary) text-(--color-text-tertiary) px-1.5 py-0.5">string</span>
      <!--v-if-->
    </div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`CodeIcon > renders with name correctly 1`] = `
"<div class="my-5">
  <div class="flex items-center gap-3 font-mono text-sm"><span class="font-semibold text-(--color-accent)">analytics</span>
    <div class="flex-1 flex items-center gap-1.5 text-xs">
      <!--v-if-->
      <!--v-if-->
    </div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`CodeIcon > renders with required correctly 1`] = `
"<div class="my-5">
  <div class="flex items-center gap-3 font-mono text-sm">
    <!--v-if-->
    <div class="flex-1 flex items-center gap-1.5 text-xs">
      <!--v-if--><span class="rounded-(--base-radius) bg-(--color-danger)/10 text-(--color-danger) px-1.5 py-0.5"> required </span>
    </div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`CodeIcon > renders with type correctly 1`] = `
"<div class="my-5">
  <div class="flex items-center gap-3 font-mono text-sm">
    <!--v-if-->
    <div class="flex-1 flex items-center gap-1.5 text-xs"><span class="rounded-(--base-radius) bg-(--color-bg-tertiary) text-(--color-text-tertiary) px-1.5 py-0.5">string</span>
      <!--v-if-->
    </div>
  </div>
  <!--v-if-->
</div>"
`;
